nsxt/resource_nsxt_policy_host_switch_profile.go

```diff
@@ -0,0 +1,592 @@
+	model.TeamingPolicy_POLICY_FAILOVER_ORDER,
+	model.TeamingPolicy_POLICY_LOADBALANCE_SRCID,
+	model.TeamingPolicy_POLICY_LOADBALANCE_SRC_MAC,
+}
+
+var overlayEncapModes = []string{
+	model.PolicyUplinkHostSwitchProfile_OVERLAY_ENCAP_VXLAN,
+	model.PolicyUplinkHostSwitchProfile_OVERLAY_ENCAP_GENEVE,
+}
+
+func resourceNsxtPolicyHostSwitchProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyHostSwitchProfileCreate,
+		Read:   resourceNsxtPolicyHostSwitchProfileRead,
+		Update: resourceNsxtPolicyHostSwitchProfileUpdate,
+		Delete: resourceNsxtPolicyHostSwitchProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtPolicyPathResourceImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"resource_type": {
+				Type:         schema.TypeString,
+				Description:  "Host switch profile type",
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(policyHostSwitchProfileResourceType, false),
+			},
+			"uplink_host_switch_profile": getUplinkHostSwitchProfileSchema(),
+		},
+	}
+}
+
+func getUplinkHostSwitchProfileSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeSet,
+		Description: "Uplink host switch profile",
+		Optional:    true,
+		MaxItems:    1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"lags": {
+					Type:        schema.TypeSet,
+					Description: "List of LACP group",
+					Optional:    true,
+					MaxItems:    64,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"id": {
+								Type:        schema.TypeString,
+								Description: "unique id",
+								Required:    false,
+								Optional:    false,
+								Computed:    true,
+							},
+							"load_balance_algorithm": {
+								Type:         schema.TypeString,
+								Description:  "LACP load balance Algorithm",
+								Required:     true,
+								ValidateFunc: validation.StringInSlice(lacpLoadBalanceAlgorithms, false),
+							},
+							"mode": {
+								Type:         schema.TypeString,
+								Description:  "LACP group mode",
+								Required:     true,
+								ValidateFunc: validation.StringInSlice(lacpGroupModes, false),
+							},
+							"name": getDataSourceRequiredStringSchema("Lag name"),
+							"number_of_uplinks": {
+								Type:         schema.TypeInt,
+								Description:  "Number of uplinks",
+								Required:     true,
+								ValidateFunc: validation.IntBetween(2, 32),
+							},
+							"timeout_type": {
+								Type:         schema.TypeString,
+								Description:  "LACP timeout type",
+								Optional:     true,
+								Default:      model.Lag_TIMEOUT_TYPE_SLOW,
+								ValidateFunc: validation.StringInSlice(lacpTimeoutTypes, false),
+							},
+							"uplinks": {
```

Comment:
The convention is to use singular noun in case element is a map - `uplink`

Reply:
Got it. Done.